flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureRequestCoordinatorTest.java

```diff
@@ -0,0 +1,311 @@
+import org.apache.flink.runtime.messages.TaskBackPressureResponse;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.ArgumentMatchers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for the {@link BackPressureRequestCoordinator}.
+ */
+public class BackPressureRequestCoordinatorTest extends TestLogger {
+
+	private static final long requestTimeout = 10000;
+	private static final double backPressureRatio = 0.5;
+
+	private static ScheduledExecutorService executorService;
+	private BackPressureRequestCoordinator coord;
+
+	@BeforeClass
+	public static void setUp() throws Exception {
+		executorService = new ScheduledThreadPoolExecutor(1);
+	}
+
+	@AfterClass
+	public static void tearDown() throws Exception {
+		if (executorService != null) {
+			executorService.shutdown();
+		}
+	}
+
+	@Before
+	public void init() throws Exception {
+		coord = new BackPressureRequestCoordinator(executorService, requestTimeout);
+	}
+
+	/** Tests simple request of task back pressure stats. */
+	@Test(timeout = 10000L)
+	public void testTriggerBackPressureRequest() throws Exception {
+		ExecutionVertex[] vertices = createExecutionVertices(ExecutionState.RUNNING, CompletionType.SUCCESSFULLY);
+
+		CompletableFuture<BackPressureStats> requestFuture = coord.triggerBackPressureRequest(vertices);
+		BackPressureStats backPressureStats = requestFuture.get();
+
+		// verify the request result
+		assertEquals(0, backPressureStats.getRequestId());
+		assertTrue(backPressureStats.getEndTime() >= backPressureStats.getStartTime());
+
+		Map<ExecutionAttemptID, Double> tracesByTask = backPressureStats.getBackPressureRatios();
+		for (ExecutionVertex executionVertex : vertices) {
+			ExecutionAttemptID executionId = executionVertex.getCurrentExecutionAttempt().getAttemptId();
+			assertEquals(backPressureRatio, tracesByTask.get(executionId), 0.0);
+		}
+
+		// verify no more pending request
+		assertEquals(0, coord.getNumberOfPendingRequests());
+
+		// verify no error on late collect
+		coord.collectTaskBackPressureStat(0, vertices[0].getCurrentExecutionAttempt().getAttemptId(), 0.0);
+	}
+
+	/** Tests simple request and collection of task back pressure stats. */
+	@Test(timeout = 10000L)
+	public void testCollectTaskBackPressureStat() throws Exception {
```

Comment:
We need to rename the test methods to better reflect the motivations.  
testTriggerBackPressureRequest -> testSuccessfulBackPressureRequest
testCollectTaskBackPressureStat -> testBackPressureRequestWithoutResponse
testRequestToExecutionFails -> testBackPressureRequestWithException